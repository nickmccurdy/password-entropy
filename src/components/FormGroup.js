import Icon from '@fortawesome/react-fontawesome'
import camelCase from 'lodash/camelCase'
import PropTypes from 'prop-types'
import React from 'react'
import { FormGroup as BootstrapFormGroup, Label } from 'reactstrap'

export default function FormGroup({ children, label, icon }) {
  return (
    <BootstrapFormGroup>
      <Label for={camelCase(label)}><Icon icon={icon} /> {label}</Label>
      {children}
    </BootstrapFormGroup>
  )
}

FormGroup.propTypes = {
  children: PropTypes.node.isRequired,
  label: PropTypes.string.isRequired,
  icon: PropTypes.oneOfType(
    [PropTypes.object, PropTypes.array, PropTypes.string].map(type => type.isRequired)
  ).isRequired
}
