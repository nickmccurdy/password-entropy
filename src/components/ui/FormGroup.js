import Icon from './Icon'
import PropTypes from 'prop-types'
import React, { PureComponent } from 'react'
import { Col, ControlLabel, FormGroup as BootstrapFormGroup } from 'react-bootstrap'

export default class FormGroup extends PureComponent {
  static propTypes = {
    children: PropTypes.node.isRequired,
    id: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    icon: PropTypes.string.isRequired
  }

  render () {
    const { children, id, label, icon } = this.props

    return (
      <BootstrapFormGroup controlId={id}>
        <Col sm={2}>
          <ControlLabel><Icon name={icon}/> {label}</ControlLabel>
        </Col>
        <Col sm={10}>{children}</Col>
      </BootstrapFormGroup>
    )
  }
}
