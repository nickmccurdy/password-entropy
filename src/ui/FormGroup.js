import PropTypes from 'prop-types'
import React from 'react'
import { Col, ControlLabel, FormGroup as BootstrapFormGroup } from 'react-bootstrap'
import Icon from 'react-fa'

export default function FormGroup ({ children, id, label, icon }) {
  return (
    <BootstrapFormGroup controlId={id}>
      <Col sm={2}>
        <ControlLabel><Icon name={icon}/> {label}</ControlLabel>
      </Col>
      <Col sm={10}>{children}</Col>
    </BootstrapFormGroup>
  )
}

FormGroup.propTypes = {
  children: PropTypes.node.isRequired,
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired
}
