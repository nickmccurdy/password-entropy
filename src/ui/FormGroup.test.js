import { shallow } from 'enzyme'
import FormGroup from './FormGroup'
import React from 'react'

it('renders', () => {
  const wrapper = shallow(
    <FormGroup id="length" label="Length" icon="arrows-h">
      Redacted
    </FormGroup>
  )

  expect(wrapper).toMatchSnapshot()
})
