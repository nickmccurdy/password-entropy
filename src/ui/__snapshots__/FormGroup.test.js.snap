// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGroup 1`] = `
<FormGroup
  bsClass="form-group"
  controlId="length"
>
  <ControlLabel
    bsClass="control-label"
    srOnly={false}
  >
    <Icon
      Component="span"
      name="arrows-h"
    />
     
    Length
  </ControlLabel>
  Redacted
</FormGroup>
`;
